test/e2e_node/e2e_service.go

```diff
@@ -235,6 +235,8 @@ func (es *e2eService) startKubeletServer() (*killCmd, error) {
 		"--v", LOG_VERBOSITY_LEVEL, "--logtostderr",
 		"--network-plugin=kubenet",
 		"--pod-cidr=10.180.0.0/24", // Assign a fixed CIDR to the node because there is no node controller.
+		"--cgroups-per-qos", "true",
```

Comment:
I don't think you need to detect systemd, do you? Can't you just check an
env var, and we'd set that manually prior to running the test? You could
also skip the test based on the value of the env var.

On Saturday, July 9, 2016, Buddha Prakash <notifications@github.com> wrote:

> In test/e2e_node/e2e_service.go
> <https://github.com/kubernetes/kubernetes/pull/27853#discussion_r70162085>
> :
>
> > @@ -235,6 +235,8 @@ func (es *e2eService) startKubeletServer() (*killCmd, error) {
> >  		"--v", LOG_VERBOSITY_LEVEL, "--logtostderr",
> >  		"--network-plugin=kubenet",
> >  		"--pod-cidr=10.180.0.0/24", // Assign a fixed CIDR to the node because there is no node controller.
> > +		"--cgroups-per-qos", "true",
>
> Oh! I completely overlooked this. I agree with you but I am not sure how
> we can achieve that. So correct me if I am wrong. To achieve this
> 1. the e2eservice needs to detect if its running on a systemd system.
> 2. if its running on a systemd system then kubelet is started without the
> cgroups-per-qos flag
> 3. We set some flag so that the cgroup hierarchy tests do not run when
> this flag is set.
> I think we can use this
> https://github.com/dubstack/kubernetes/blob/0a7b6cd2c88f561f4d8c3fb9a095d522ec33dc6d/test/e2e_node/e2e_service.go#L211
> for first point
> But I am not sure how to handle point 3.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27853/files/0a7b6cd2c88f561f4d8c3fb9a095d522ec33dc6d#r70162085>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAABYhAFx2zigzELSt_gjTKdwrzfnm-zks5qTx3fgaJpZM4I7o6F>
> .
>